### ISSUE TITLE
Does the shape of the simulation box has to be a cube on periodic boundary conditions

### ISSUE THREAD

**Issue by hkayabilisim** [N-BodyShop/changa#17]
Hi,

When I look at the source codes of Ewald implementation here:
https://github.com/N-BodyShop/changa/blob/27cbc83121ada9a2c01d0869802c77f49bfd4000/Ewald.cpp#L123

I see that the simulation box is assumed to be a cube. Does it mean that I cannot use rectangular boxes with different  sidelengths in the periodic boundary conditions?

Thanks
Huseyin

**Comment by trquinn** [comment]
You are correct: with Ewald gravity, the periodic box is assumed to be a cube.  Without Ewald, (say a non-self gravitating turbulent box) the period box does not have to be a cube.

I've never seen an implementation of a non-cubical Ewald,  I would imagine that you would want the real space/k space transition parameter, alpha, to depend on direction in that case.  A lot of care and testing would be needed.
